apps/dashboard/src/app/(app)/team/[team_slug]/[project_slug]/(sidebar)/assets/create/nft/create-nft-page.tsx

```diff
@@ -0,0 +1,502 @@
+"use client";
+import { revalidatePathAction } from "@/actions/revalidate";
+import { useTrack } from "hooks/analytics/useTrack";
+import { useRef } from "react";
+import {
+  type ThirdwebClient,
+  defineChain,
+  encode,
+  getContract,
+  sendAndConfirmTransaction,
+} from "thirdweb";
+import { deployERC721Contract, deployERC1155Contract } from "thirdweb/deploys";
+import { multicall } from "thirdweb/extensions/common";
+import {
+  lazyMint as lazyMint721,
+  setClaimConditions as setClaimConditions721,
+} from "thirdweb/extensions/erc721";
+import {
+  getNFTs as getNFTs1155,
+  lazyMint as lazyMint1155,
+  setClaimConditions as setClaimConditions1155,
+} from "thirdweb/extensions/erc1155";
+import { useActiveAccount } from "thirdweb/react";
+import { maxUint256 } from "thirdweb/utils";
+import { parseError } from "utils/errorParser";
+import { useAddContractToProject } from "../../../hooks/project-contracts";
+import type {
+  CreateNFTFormValues,
+  NFTCollectionInfoFormValues,
+} from "./_common/form";
+import {
+  getNFTDeploymentTrackingData,
+  getNFTStepTrackingData,
+} from "./_common/tracking";
+import { CreateNFTPageUI } from "./create-nft-page-ui";
+import type { NFTMetadataWithPrice } from "./upload-nfts/batch-upload/process-files";
+
+export function CreateNFTPage(props: {
+  accountAddress: string;
+  client: ThirdwebClient;
+  teamSlug: string;
+  projectSlug: string;
+  teamId: string;
+  projectId: string;
+}) {
+  const activeAccount = useActiveAccount();
+  const trackEvent = useTrack();
+  const addContractToProject = useAddContractToProject();
+  const contractAddressRef = useRef<string | undefined>(undefined);
+
+  function getContractInfo(params: {
+    chain: string;
+  }) {
+    if (!activeAccount) {
+      throw new Error("Wallet is not connected");
+    }
```

Comment:
why do we need a connected wallet to get contract info?

Reply:
function name is misleading, renamed it